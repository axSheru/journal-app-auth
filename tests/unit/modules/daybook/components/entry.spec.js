import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry'
import { journalState } from '../../../mock-data/test-journal-state'

describe('Pruebas en el entry component.', () => {

    const mockRouter = {
        push: jest.fn()
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount( Entry, {
            props: {
                entry: journalState.entries[0]
            },
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })
    });

    test('Debe de hacer match con el snapshot.', () => {
        
        expect( wrapper.html() ).toMatchSnapshot()

    })

    test('Debe de redireccionar al hacer click en el entry-container.', () => {
        
        wrapper.find('.entry-container').trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'entry', params: { id: journalState.entries[0].id } })

    })

    test('Pruebas en las propiedades computadas.', () => {
        
        expect(wrapper.vm.day).toBe(18)
        expect(wrapper.vm.month).toBe('Septiembre')
        expect(wrapper.vm.yearDay).toBe('2021, SÃ¡bado')

    })
    
})
