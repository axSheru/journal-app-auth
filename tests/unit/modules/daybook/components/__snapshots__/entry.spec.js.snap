// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el entry component. Debe de hacer match con el snapshot. 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Título -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">18</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-2 fw-light">2021, Sábado</span></div>
  <div class="entry-description">The mandalorian caughts the Mythrol.</div>
</div>
`;
